node-js-basics-101/node_modules/ws/lib/PerMessageDeflate.js

```diff
@@ -0,0 +1,384 @@
+'use strict';
+
+const safeBuffer = require('safe-buffer');
+const zlib = require('zlib');
+
+const bufferUtil = require('./BufferUtil');
+
+const Buffer = safeBuffer.Buffer;
+
+const AVAILABLE_WINDOW_BITS = [8, 9, 10, 11, 12, 13, 14, 15];
+const TRAILER = Buffer.from([0x00, 0x00, 0xff, 0xff]);
+const EMPTY_BLOCK = Buffer.from([0x00]);
+const DEFAULT_WINDOW_BITS = 15;
+const DEFAULT_MEM_LEVEL = 8;
```

Comment:
__[jshint]__
* 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

<!-- Comment by [SideCI](https://sideci.com) -->